jsonnet/main.jsonnet

```
@@ -123,6 +123,36 @@ local inCluster =
         },
         kubeRbacProxyImage: $.values.common.images.kubeRbacProxy,
         promLabelProxyImage: $.values.common.images.promLabelProxy,
+        config: {
+          global: {
+            resolve_timeout: '5m',
+          },
+          inhibit_rules: [{
+            source_matchers: ['severity = critical'],
+            target_matchers: ['severity =~ warning|info'],
+            equal: ['namespace', 'alertname'],
+          }, {
+            source_matchers: ['severity = warning'],
+            target_matchers: ['severity = info'],
+            equal: ['namespace', 'alertname'],
+          }],
+          route: {
+            group_by: ['namespace'],
+            group_wait: '30s',
+            group_interval: '5m',
+            repeat_interval: '12h',
+            receiver: 'Default',
+            routes: [
+              { receiver: 'Watchdog', matchers: ['alertname = Watchdog'] },
+              { receiver: 'Critical', matchers: ['severity = critical'] },
+            ],
+          },
+          receivers: [
+            { name: 'Default' },
+            { name: 'Watchdog' },
+            { name: 'Critical' },
+          ],
+        },
```

How about just combining `config` and `route` with `+`? This way we'll still get new fields + updates for the others fields from upstream.
(Of course we may get "broken/incomplete" updates as we'll be overwriting some of the fields...)

```suggestion
        config+: {
          inhibit_rules: [{
            source_matchers: ['severity = critical'],
            target_matchers: ['severity =~ warning|info'],
            equal: ['namespace', 'alertname'],
          }, {
            source_matchers: ['severity = warning'],
            target_matchers: ['severity = info'],
            equal: ['namespace', 'alertname'],
          }],
          route+: {
            routes: [
              { receiver: 'Watchdog', matchers: ['alertname = Watchdog'] },
              { receiver: 'Critical', matchers: ['severity = critical'] },
            ],
          },
          receivers: [
            { name: 'Default' },
            { name: 'Watchdog' },
            { name: 'Critical' },
          ],
        },
```

---

I'm on the fence: I find it clearer to have the full config here than reviewing the merged config. Again the configuration provisioned on cluster creation is only informative.

---

I had a feeling :)